libminifi/src/core/Processor.cpp

```diff
@@ -386,6 +379,38 @@ std::shared_ptr<Connectable> Processor::pickIncomingConnection() {
   return getNextIncomingConnectionImpl(rel_guard);
 }
 
+void Processor::validateAnnotations() const {
+  switch (getInputRequirement()) {
+    case annotation::Input::INPUT_REQUIRED: {
+      if (!hasIncomingConnections()) {
+        throw Exception(PROCESS_SCHEDULE_EXCEPTION, "INPUT_REQUIRED was specified for the processor, but no inputs were found");
```

Comment:
I would rather refer to incoming connections in the error message.
```suggestion
        throw Exception(PROCESS_SCHEDULE_EXCEPTION, "INPUT_REQUIRED was specified for the processor, but no incoming connections were found");
```